src/scm/github.js

```diff
@@ -312,156 +335,169 @@ class Github {
-      Object.assign(context.bindBranch, { [context.id] : gist });
-      chrome.storage.sync.set({ bindBranch: context.bindBranch });
-      return response;
-    })
-    .catch(err => {
-      throw new Error('Failed to create new gist.');
-    });
+      .then(response => {
+        const gist = response.id;
+        context.branch = gist;
+        Object.assign(context.bindBranch, {
+          [context.id]: gist
+        });
+        chrome.storage.sync.set({
+          bindBranch: context.bindBranch
+        });
+        return response;
+      })
+      .catch(err => {
+        throw new Error('Failed to create new gist.');
+      });
   }
 
   createBranch() {
     const branch = $('#new-branch-name').val();
     if (!branch || branch === '') return;
     return new Promise((resolve, reject) => {
-      $.getJSON(
-        `${this.baseUrl}/repos/${context.repo.fullName}/git/refs/heads/${context.branch}`,
-        { access_token: this.accessToken }
-      )
-      .then(resolve)
-      .fail(reject)  
-    })
-    .then(response => {
-      if (response.object) {
-        return response.object.sha;
-      }
-      else {
-        return $.getJSON(
-          `${this.baseUrl}/repos/${context.repo.fullName}/git/refs/heads`,
-          { access_token: this.accessToken }
-        )
-        .then(response => {
-          return response[0].object.sha;
-        })
-      }
-    })
-    .then(sha => {
-      const payload = {
-        ref: `refs/heads/${branch}`,
-        sha: sha
-      };
-      return $.ajax({
-        url: `${this.baseUrl}/repos/${context.repo.fullName}/git/refs`,
-        headers: {
-          'Authorization': `token ${this.accessToken}`
-        },
-        method: 'POST',
-        crossDomain: true,
-        dataType: 'json',
-        contentType: 'application/json',
-        data: JSON.stringify(payload)
+        getGitHubJSON(
+          `${this.baseUrl}/repos/${context.repo.fullName}/git/refs/heads/${context.branch}`,
+          this.accessToken)
+          .then(resolve)
+          .fail(reject)
+      })
+      .then(response => {
+        if (response.object) {
+          return response.object.sha;
+        } else {
+          return getGitHubJSON(
+            `${this.baseUrl}/repos/${context.repo.fullName}/git/refs/heads`,
+            this.accessToken)
+            .then(response => {
+              return response[0].object.sha;
+            })
+        }
+      })
+      .then(sha => {
+        const payload = {
+          ref: `refs/heads/${branch}`,
+          sha: sha
+        };
+        return $.ajax({
+          url: `${this.baseUrl}/repos/${context.repo.fullName}/git/refs`,
+          headers: {
+            'Authorization': `token ${this.accessToken}`
+          },
+          method: 'POST',
+          crossDomain: true,
+          dataType: 'json',
+          contentType: 'application/json',
+          data: JSON.stringify(payload)
+        });
+      })
+      .then(response => {
```

Comment:
<table data-meta="generated_by_danger">
  <tbody>
    <tr>
      <td>:warning:</td>
      <td width="100%" data-sticky="false"><span data-href="https://github.com/leonhartX/gas-github/blob/15f4157afeba0560f3f12968d583a6f4fceda1cf/src/scm/github.js#L464"></span>‘response’ is defined but never used.</td>
    </tr>
  </tbody>
</table>